### DISCUSSION TITLE
List of dependencies when installing sharpy without conda?

### DISCUSSION THREAD

**Discussion by samfux84** [ImperialCollegeLondon/sharpy#139] [Q&A]
Hi,

On many HPC clusters, conda is not supported, because it creates a large number of small files (a simple conda installation can easily accumulate 100k-200k files with an average filesize of KB), which does have a negative impact on the performance of parallel high-performance file systems like Lustre. Apart from the Lustre file system, our cluster also has some home directories for the users, but there we have a inode quota of 100k (with more than 3000 users, this is a must, as we have a nightly backup of all home directories and therefore we need to limit the total number of files as the backup needs to be finished within 24 hours) and therefore users cannot install conda locally in their home directory.

But we have a large portfolio of centrally installed software:

https://scicomp.ethz.ch/wiki/Euler_applications_and_libraries

I think it would be feasible for us to install sharpy centrally without using conda. Can you please provide a list of the required dependencies (not conda packages, not OS rpms, just the libraries that are required)?

Best regards

Sam

**Comment by ngoiz** [comment]
Hi Sam,

Unfortunately, we haven't really used SHARPy on HPC environments hence our use of conda. We don't have a list of required dependencies so my best suggestion would be to go through the `environment_minimal.yml` which is the most lightweight option we have with minimal libraries.

Alternatively, a brute force approach to gather the minimal libraries might be to search through the code for any `import X` calls.

Hope this helps,

Norberto

(from experience, the most troublesome library we have is `tvtk` which is used to plot to paraview by our `AerogridPlot` and `BeamPlot` post-processors, so if you are not looking to save the paraview files on your HPC set-up you may not need those).
